src/utils/hooks/useGetQuestionReviewHistory.tsx

```diff
@@ -35,15 +44,22 @@ const useGetQuestionReviewHistory = ({ isApplicant, ...variables }: UseGetQuesti
     const { applicationResponses, reviewResponses } =
       data?.templateElementByTemplateCodeAndCodeAndTemplateVersion as TemplateElement
 
-    const allResponses: ResponsesByDate = {}
+    const allResponsesByStage: ResponsesByStageByDate = {}
 
     applicationResponses.nodes.forEach((applicantResponse) => {
       if (!applicantResponse) return
-      const { timeUpdated, application, value } = applicantResponse
+      const { timeUpdated, stageNumber, application, id, value } = applicantResponse
       const { firstName, lastName } = application?.user as User
 
+      if (!stageNumber) {
+        console.log(`application_reponse ${id} without stage_number: not added to History`)
+        return
```

Comment:
If there's a response with a missing stage number (obviously it shouldn't happen), then maybe we should still display it in the UI (with maybe a "NO STAGE" label?), otherwise the user will never know anything is wrong.

Reply:
Hmmm ok. That's a good point. Will work on that after break. Thanks

Reply:
Fixed: 1c999d0